### PR TITLE
The SCP-409 (funni crystal) Chronicles

### PR DESCRIPTION
**Pull request by 77seven77seven**
## About the Pull Request

i just wanted SCP-409 in the game :)

## Why It's Good For The Game

SCP-409 is one of the many iconic SCPs and it would be a shame if it weren't in, also it's funni.

## Changelog

:cl:
add: Added the SCP-409 Containment Chamber near the SCP-500 Containment chamber.
add: Added SCP-409 itself.
/:cl:

### CODE REVIEW

**Comment by 77seven77seven** [comment]
Pictures to accompany the chaos:
image
image

**Comment by Okram0** [comment]
Can you describe how it works mechanically speaking?

**Comment by 77seven77seven** [comment]
> Can you describe how it works mechanically speaking?

You click the crystal, you touch the crystal. When you touch the crystal it takes 60 BYOND ticks to turn you into crystal residue (which doesn't function the same as the crystal for obvious reasons.)

File: baystation12.dme [hunk 1]
```diff
@@ -3418,6 +3419,7 @@
 #include "maps\site53\items\bullets.dm"
 #include "maps\site53\items\clothes.dm"
 #include "maps\site53\items\guns.dm"
+#include "maps\site53\items\manuals.dm"
```
Comment (Tsurupeta) [inline baystation12.dme]: bruh

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/obj/structure/scp409/
+	name = "SCP-409"
+	desc = "A white crystal, something beckons you to come and touch it..."
+	icon = 'icons/SCP/scp-409.dmi'
+	icon_state = "409"
+	layer = 2.9
+	anchored = TRUE // makes it not able to be moved
+	density = TRUE // makes it have collisions
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: why so many useless comments

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	SCP = /datum/scp/scp_409
+
+/datum/scp/scp_409
+	name = "SCP-409"
+	designation = "409"
+	classification = KETER
+
+/obj/effect/decal/cleanable/residue409
+	name = "crystals"
+	desc = "Some crystals on the floor... is that blood?" // something about blood even though it doesnt have blood on the sprite, blame the spriting team
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: why so many useless comments

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	icon = 'icons/SCP/scp-409.dmi'
+	icon_state = "residue409"
+	gender = PLURAL
+
+/obj/structure/scp409/attack_hand(mob/user as mob)
+	visible_message("[user] touches [src].")
+	log_and_message_admins("[user] has touched SCP-409 at [x] [y] [z]") // not sure if co-ordinates are required for the JUMP option to be available but its cool :sunglasses:
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: why so many useless comments

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	Crystal(user)
+
+/obj/structure/scp409/proc/Crystal(var/mob/living/user) // defines Crystal() which calls crystaltim(), i am dumb
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: why so many useless comments

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	if(istype(user, /mob/living/carbon/human/scp049))
+		to_chat(SPAN_WARNING("Your doctor instincts prevent you from touching that odd crystal."))
+	else if(istype(user))
+		sleep(60) // makes it so that it takes a while to take effect
+		playsound(src, 'sound/voice/emotes/male_scream2.ogg', 25) // I didn't find any other solution to do this, so this sound will play regardless if the victim is male or female. TODO: fix dis
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: why so many useless comments

File: code/modules/mob/death.dm [hunk 7]
```diff
@@ -46,6 +46,30 @@
 	remove_from_dead_mob_list()
 	addtimer(CALLBACK(src, .proc/check_delete, animation), 15)
 
+/mob/proc/crystaltim(anim="h2crystal",remains=/obj/effect/decal/cleanable/residue409)
```
Comment (Tsurupeta) [inline code/modules/mob/death.dm]: dumb proc names

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+		log_and_message_admins("[user] has died from SCP-409 exposure at [x] [y] [z]")
+		user.visible_message(
+			SPAN_DANGER(FONT_LARGE("You scream, as your body starts to twist and turn, it's flesh being replaced by rock. Stinging into your skin, you try and scream but the crystals already covered your mouth.")),
+			SPAN_WARNING("[user] screams and starts to turn into crystals on the floor.")
+		)
+		user.crystaltim() // TODO: make it use CA (crystal agent)
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: wheres newline?

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 5 above)
Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-409.dm]: Also, proc name shouldn't start with capital letter

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	classification = KETER
+
+/obj/effect/decal/cleanable/residue409
+	name = "crystals"
+	desc = "Some crystals on the floor... is that blood?" // something about blood even though it doesnt have blood on the sprite, blame the spriting team
+	icon = 'icons/SCP/scp-409.dmi'
+	icon_state = "residue409"
+	gender = PLURAL
+
+/obj/structure/scp409/attack_hand(mob/user as mob)
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: why `as mob`? It's not needed

**Review by Tsurupeta — CHANGES_REQUESTED** [review]

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 2 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: new coder friendliness :) also its just comments, why do you care?

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 3 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: new coder friendliness :) also its just comments, why do you care?

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 4 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: new coder friendliness :) also its just comments, why do you care?

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 5 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: I'll fix that.

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 6 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: it's a TODO

File: code/modules/mob/death.dm (hunk 7 above)
Comment (77seven77seven) [reply code/modules/mob/death.dm]: sowwy

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 8 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: gone

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 9 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: i stole SM code

File: baystation12.dme (hunk 1 above)
Comment (77seven77seven) [reply baystation12.dme]: guh whats wrong

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 2 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: comment issues fixed

File: code/modules/mob/death.dm (hunk 7 above)
Comment (77seven77seven) [reply code/modules/mob/death.dm]: fixed

File: baystation12.dme (hunk 1 above)
Comment (Tsurupeta) [reply baystation12.dme]: duplicate definition

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 9 above)
Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-409.dm]: remove it

File: baystation12.dme (hunk 1 above)
Comment (Tsurupeta) [reply baystation12.dme]: image

File: code/modules/SCP/SCPs/SCP-409.dm [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+		to_chat(SPAN_WARNING("Your doctor instincts prevent you from touching that odd crystal."))
+	else if(istype(user))
+		sleep(60)
+		playsound(src, 'sound/voice/emotes/male_scream2.ogg', 25) // I didn't find any other solution to do this, so this sound will play regardless if the victim is male or female. TODO: fix dis
+		log_and_message_admins("[user] has died from SCP-409 exposure at [x] [y] [z]")
+		user.visible_message(
+			SPAN_DANGER(FONT_LARGE("You scream, as your body starts to twist and turn, it's flesh being replaced by rock. Stinging into your skin, you try and scream but the crystals already covered your mouth.")),
+			SPAN_WARNING("[user] screams and starts to turn into crystals on the floor.")
+		)
+		user.crystallize() // this is the thing that kills the person. TODO: make it use CA (crystal agent)
```
Comment (Tsurupeta) [inline code/modules/SCP/SCPs/SCP-409.dm]: still no newline at the end of file

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 9 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: done

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 10 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: it should be there now i domnt fucking know

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 8 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: okay it should be there now

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 10 above)
Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-409.dm]: no its not

**Comment by Tsurupeta** [comment]
image
wtf is this file

**Comment by 77seven77seven** [comment]
> image wtf is this file

a shortcut for build.cmd

**Comment by Tsurupeta** [comment]
delete it

**Comment by 77seven77seven** [comment]
> delete it

ok

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 10 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: > no its not

HOW THRE FUCK DO I ADD TI THEN

Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-409.dm]: use vsc instead of dm

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: > use vsc instead of dm

i dont have money for that

Comment (Tsurupeta) [reply code/modules/SCP/SCPs/SCP-409.dm]: It's free

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: my pc can't even handle it

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: okay im fucking using it, i dont see any way to fix YOUR FUCKING NEW LINE ISSUE

**Comment by Okram0** [comment]
I feel like the residue should do the same thing if you walk on it because it IS keter class and I assume this is in HCZ and thus hard to access but idk it's your code.

**Comment by 77seven77seven** [comment]
> I feel like the residue should do the same thing if you walk on it because it IS keter class and I assume this is in HCZ and thus hard to access but idk it's your code.

not going to do that for ze funni balance changes

File: code/modules/SCP/SCPs/SCP-409.dm (hunk 10 above)
Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: okay i fixed it

Comment (77seven77seven) [reply code/modules/SCP/SCPs/SCP-409.dm]: or i at least think, until now I'm gonna mark this as resolved

**Comment by aidan-bowman** [comment]
> I feel like the residue should do the same thing if you walk on it because it IS keter class and I assume this is in HCZ and thus hard to access but idk it's your code.

Hard agree.

It might seem overpowered to have different ways for the crystals to spread, but if you keep it the way it is now (touch the crystal and die) 409 will end up like all the other useless skips that people forget even exists. As long as causing the initial outbreak is rare (and admins bwoink anyone repeatedly griefing with it), I don't see why it can't act as an actual threat.

**Comment by 77seven77seven** [comment]
> > I feel like the residue should do the same thing if you walk on it because it IS keter class and I assume this is in HCZ and thus hard to access but idk it's your code.
> 
> Hard agree.
> 
> It might seem overpowered to have different ways for the crystals to spread, but if you keep it the way it is now (touch the crystal and die) 409 will end up like all the other useless skips that people forget even exists. As long as causing the initial outbreak is rare (and admins bwoink anyone repeatedly griefing with it), I don't see why it can't act as an actual threat.

eh, you're right. ill make it that you need to click the 409 crystal but you just needing to touch the crystal residue for it to spread

**Comment by 77seven77seven** [comment]
Closed due to the fact that I still have a long way ahead with my coding and spriting abilities.

**Pull request closed without merging**